frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -625,105 +681,136 @@ pub mod pallet {
             ledger.unbonding_info = future_chunks;
 
             Self::update_ledger(&staker, ledger);
+            EraRewardsAndStakes::<T>::mutate(&current_era, |value| {
+                if let Some(x) = value {
+                    x.locked = x.locked.saturating_add(withdraw_amount)
+                }
+            });
 
             Self::deposit_event(Event::<T>::Withdrawn(staker, withdraw_amount));
 
             Ok(().into())
         }
 
-        /// Claim the rewards earned by contract_id.
-        /// All stakers and developer for this contract will be paid out with single call.
-        /// claim is valid for all unclaimed eras but not longer than history_depth().
-        /// Any reward older than history_depth() will go to Treasury.
-        /// Any user can call this function.
-        #[pallet::weight(T::WeightInfo::claim(T::MaxNumberOfStakersPerContract::get() + 1))]
-        pub fn claim(
+        // TODO: do we need to add force methods or at least methods that allow others to claim for someone else?
```

Comment:
once we implement re-stake, projects might be interested to claim in name of the stakers and thus increase the staked amount for their contracts

Reply:
Sure, I was thinking the same thing.

It's easy to add a wrapped call, no problem there.

Reply:
I can implement wrappers after I get review from Aleks.